test/Output.hs

```diff
@@ -0,0 +1,223 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+-- | This module provides some simple pretty-printing output verification tests.
+-- This amounts to spot-checking a few places in the pretty printing that are
+-- examined by hand: in general, it its much more effective to do round-trip
+-- testing using AST's parsed from bitcode generated by actual programs, which is
+-- what occurs in the llvm-pretty-bc-parser package (i.e. much more comprehensive
+-- testing is deferred to the llvm-pretty-bc-parser package, which might then
+-- reveal issues that need to be fixed in this llvm-pretty package).
+
+module Output ( tests ) where
+
+import qualified Data.Text as T
+import qualified Test.Tasty as Tasty
+import           Test.Tasty.HUnit
+import qualified Text.PrettyPrint as PP
+
+import Text.LLVM.AST
+import Text.LLVM.PP
+
+import TQQDefs
+
+tests :: Tasty.TestTree
+tests = Tasty.testGroup "LLVM pretty-printing output tests"
+  $ let -- s1 is a non-sensical construct whose primary intention is to hold two
+        -- sub-structures that change their pretty representations at different
+        -- LLVM versions.  The pretty output will be checked at different LLVM
+        -- versions to ensure that the desired version-specific changes in the
+        -- output are seen.
+        s1 = Effect
+             (GEP True (Alias (Ident "hi")) (Typed Opaque dcu) [])
+             [] :: Stmt
```

Comment:
Nit pick: I'd prefer putting the type signature up front:

```hs
s1 :: Stmt
s1 = Effect
     (GEP ...)
     []
```

Reply:
Updated.